js/test/test.js

```diff
@@ -103,51 +107,96 @@ for (const [credential, isRequired] of Object.entries (requiredCredentials)) {
 }
 
 if (settings && settings.skip) {
-    console.log ('[Skipped]', { 'exchange': exchangeId, 'symbol': exchangeSymbol || 'all' })
-    process.exit ()
+    dump ('[Skipped]', { 'exchange': exchangeId, 'symbol': exchangeSymbol || 'all' })
+    process.exit ();
 }
 
-//-----------------------------------------------------------------------------
+async function runMethod(exchange, methodName, ... args) {
+    return await tests[methodName] (exchange, ... args);
+}
 
-async function test (methodName, exchange, ... args) {
-    console.log ('Testing', exchange.id, methodName, '(', ... args, ')')
+function testMethodAvailableForCurrentLang(methodName) {
+    return tests[methodName] !== undefined;
+}
+
+function assertMethod (condition, message) {
+    assert (condition, message);
+}
+
+function getIncludedSymbols (exchange, symbols) {
+    return exchange.symbols.filter ((symbol) => symbols.indexOf (symbol) >= 0);
+}
+
+function getMarketsForCode (markets, code) {
+    return markets.filter ((market) => (market['base'] === code));
+}
+
+function getSymbolsFromMarkets (markets) {
+    return markets.map (market => market['symbol']);
+}
+
+function getActiveMarkets (markets) {
+    return markets.filter ((market) => !exchange.safeValue (market, 'active', false));
+}
+
+function findValueIndexInArray (arr, value) {
+    return arr.indexOf (value);
+}
+
+function exceptionHint (exc) {
+    return '[' + exceptionTypeName (exc.constructor.name) + '] ' + exc.message.slice (0, 200);
+}
+
+// ----------------------------------------------------------------------------
+// ### AUTO-TRANSPILER-START ###
+// ----------------------------------------------------------------------------
```

Comment:
all things between `AUTO-TRANSPILER-START` and ...END will be transpiler later. 
also, note that i've added semicolon `;` in the end of all lines down below.